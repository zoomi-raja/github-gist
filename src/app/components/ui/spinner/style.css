.load {
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.5);
  position: absolute;
  top: 0;
  left: 0;
}
.loader {
	border: 5px solid #f3f3f3;
	-webkit-animation: spin 1s linear infinite;
	animation: spin 1s linear infinite;
	border-top: 5px solid #777;
	border-radius: 50%;
	width: 30px;
	height: 30px;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}
@-webkit-keyframes spin {
	0% {
		-webkit-transform: rotate(0deg);
	}
	100% {
		-webkit-transform: rotate(360deg);
	}
}

@keyframes spin {
	0% {
		transform: rotate(0deg);
	}
	100% {
		transform: rotate(360deg);
	}
}
