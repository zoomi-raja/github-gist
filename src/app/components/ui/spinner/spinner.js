import './style.css';

export const Spinner = () => {
	return <div className='load'><div className='loader'></div></div>;
};