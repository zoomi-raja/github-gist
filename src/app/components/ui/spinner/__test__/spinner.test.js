import renderer from 'react-test-renderer';
import { Spinner } from '../spinner';

it('spinner snapshot', () => {
  const tree = renderer.create(<Spinner />).toJSON();
  expect(tree).toMatchSnapshot();
})