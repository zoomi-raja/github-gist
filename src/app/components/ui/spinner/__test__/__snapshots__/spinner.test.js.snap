// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spinner snapshot 1`] = `
<div
  className="load"
>
  <div
    className="loader"
  />
</div>
`;
